### PR TITLE
Fix: report flow_rate=0 when pump is not running

### PR DESCRIPTION
**Pull request by cy-bertrand**
The gateway reports a default/calibration value of 100 L/h for flow_rate even when the pump is stopped. This mirrors the fix i've already applied in the Home Assistant integration (cloudless-garden/ha-gardena-smart-local-preview#113) by using the existing is_running property to short-circuit and return 0 when the pump is confirmed off (while still returning None if the running state hasn't been read yet.)

Should resolve the issue https://github.com/cloudless-garden/gardena-smart-local-api/issues/93

### CODE REVIEW

File: gardena_smart_local_api/devices/irrigation.py [hunk 1]
```diff
@@ -518,9 +518,14 @@ def outlet_temperature_min(self) -> int | None:
         if isinstance(value, int):
             return value
         return None
-
+
```
Comment (rettichschnidi) [inline gardena_smart_local_api/devices/irrigation.py]: Unrelated change, please undo.

File: gardena_smart_local_api/devices/irrigation.py [hunk 2]
```diff
@@ -518,9 +518,14 @@ def outlet_temperature_min(self) -> int | None:
         if isinstance(value, int):
             return value
         return None
-
+        
     @property
     def flow_rate(self) -> int | None:
+    # The gateway reports flow_rate=100 as a default/calibration value
+    # even when the pump is stopped. Only report a real flow when the
+    # pump is actually running.
```
Comment (rettichschnidi) [inline gardena_smart_local_api/devices/irrigation.py]: I think this is a bit more useful:

```suggestion
    # The device can report values from 100 to 13615. Therefore, even when the pump
    # is not running, a value of at least 100 gets reported.
    # The following condition avoids passing on obviously wrong values.
```

**Review by rettichschnidi — CHANGES_REQUESTED** [review]

File: gardena_smart_local_api/devices/irrigation.py (hunk 2 above)
Comment (cy-bertrand) [reply gardena_smart_local_api/devices/irrigation.py]: Ok.
Actually not at "least 100" but exactly 100 😜

**Review by rettichschnidi — CHANGES_REQUESTED** [review]
Thanks!

Now please merge those two commits into one and make sure to follow the gitlint findings (CI has some complains).

If you do not know how to do this let us know. We'll either explain it or will do it for you.

**Comment by easybe** [comment]
> Thanks!
> 
> Now please merge those two commits into one and make sure to follow the gitlint findings (CI has some complains).
> 
> If you do not know how to do this let us know. We'll either explain it or will do it for you.

Some hints are given here: https://github.com/cloudless-garden/gardena-smart-local-api#commits

File: gardena_smart_local_api/devices/irrigation.py [hunk 3]
```diff
@@ -521,6 +521,11 @@ def outlet_temperature_min(self) -> int | None:
 
     @property
     def flow_rate(self) -> int | None:
+    # The device can report values from 100 to 13615. Therefore, even when the pump
+    # is not running, a value of 100 gets reported.
+    # The following condition avoids passing on obviously wrong values.
+        if self.is_running is False:
```
Comment (easybe) [inline gardena_smart_local_api/devices/irrigation.py:527]:
```suggestion
        if not self.is_running:
```

File: gardena_smart_local_api/devices/irrigation.py [hunk 4]
```diff
@@ -521,6 +521,11 @@ def outlet_temperature_min(self) -> int | None:
 
     @property
     def flow_rate(self) -> int | None:
+    # The device can report values from 100 to 13615. Therefore, even when the pump
+    # is not running, a value of 100 gets reported.
+    # The following condition avoids passing on obviously wrong values.
```
Comment (easybe) [inline gardena_smart_local_api/devices/irrigation.py]:
```suggestion
        # The device can report values from 100 to 13615. Therefore, even when the pump
        # is not running, a value of 100 gets reported.
        # The following condition avoids passing on obviously wrong values.
```

**Review by easybe — CHANGES_REQUESTED** [review]

File: gardena_smart_local_api/devices/irrigation.py (hunk 3 above)
Comment (cy-bertrand) [reply gardena_smart_local_api/devices/irrigation.py:527]: Hi @easybe,

If I'm not wrong if we put "if not self.is_running:" then the value will report 0 instead of "none" when the device status is not know. Am I'm wrong ?

I'd rather not assert a fact (zero flow) we haven't actually verified. So I'll keep is False here — let me know if you see a case where that causes trouble.

Comment (easybe) [reply gardena_smart_local_api/devices/irrigation.py:527]: Yes, you are right, did not think of that.

**Comment by cy-bertrand** [comment]
ok i've tried to squash the commits into one and change the text to be gitlints compliant

**Review by rettichschnidi — CHANGES_REQUESTED** [review]
Would delete the following in the Git commit, because it is not relevant for the history of this repository and that PR will not be merged:

> This mirrors the fix already applied in the Home Assistant integration (https://github.com/cloudless-garden/ha-gardena-smart-local-preview/pull/113) by using the existing is_running property to short-circuit and return 0 when the pump is confirmed off (while still returning None if the running state hasn't been read yet). 

Everything else LGTM.

**Review by rettichschnidi — APPROVED** [review]

**Comment by easybe** [comment]
Tests need to be adapted too...

**Comment by cy-bertrand** [comment]
> Tests need to be adapted too...

**Suggestion for the test**

Since the new logic overrides `flow_rate` to `0` when the pump is stopped, the test should reflect that behavior. The fixture can stay as-is (with the device reporting `flow_rate: 100` while `pump_state: 0` — a realistic edge case), and we just update the expected result:

In `test_epp.py`, change:

```python
assert epp.flow_rate == 100
```

to:

```python
assert epp.flow_rate == 0
```

This validates the fix: even when the underlying device reports `100`, the computed `flow_rate` correctly returns `0` because the pump is idle.

**Comment by rettichschnidi** [comment]
@cy-bertrand Rather than suggesting fixes, you should implement them and get CI to pass. Without, the pipeline fails and we can not merge this PR.

**Review by kelvan — APPROVED** [review]
lgtm

**Review by easybe — APPROVED** [review]
